apps/server/src/modules/learnroom/service/board-copy.service.ts

```diff
@@ -23,53 +27,107 @@ export type BoardCopyParams = {
-			const element = boardElements[i];
+		const { originalBoard, user, destinationCourse } = params;
 
-			if (element.boardElementType === BoardElementType.Task) {
-				const originalTask = element.target as Task;
-				const status = this.taskCopyService.copyTaskMetadata({
-					originalTask,
-					user: params.user,
-					destinationCourse: params.destinationCourse,
-				});
-				elements.push(status);
-				const taskBoardElement = BoardElement.FromTask(status.copyEntity as Task);
-				references.push(taskBoardElement);
-			} else if (element.boardElementType === BoardElementType.Lesson) {
-				const originalLesson = element.target as Lesson;
-				// eslint-disable-next-line no-await-in-loop
-				const status = await this.lessonCopyService.copyLesson({
-					originalLesson,
-					user: params.user,
-					destinationCourse: params.destinationCourse,
-				});
-				elements.push(status);
-				const lessonBardElement = BoardElement.FromLesson(status.copyEntity as Lesson);
-				references.push(lessonBardElement);
-			}
-		}
+		const boardElements: BoardElement[] = originalBoard.getElements();
+		const elements: CopyStatus[] = await this.copyBoardElements(boardElements, user, destinationCourse);
+		const references: BoardElement[] = this.extractReferences(elements);
 
-		const copy = new Board({ references, course: params.destinationCourse });
-		const status = {
+		let boardCopy: Board = new Board({ references, course: destinationCourse });
+		let status: CopyStatus = {
 			title: 'board',
 			type: CopyElementType.BOARD,
 			status: this.copyHelperService.deriveStatusFromElements(elements),
-			copyEntity: copy,
+			copyEntity: boardCopy,
 			originalEntity: params.originalBoard,
 			elements,
 		};
 
+		status = this.updateCopiedEmbeddedTasksOfLessons(status);
+		if (status.copyEntity) {
+			boardCopy = status.copyEntity as Board;
+		}
+		await this.boardRepo.save(boardCopy);
+
 		return status;
 	}
+
+	private async copyBoardElements(
+		boardElements: BoardElement[],
+		user: User,
+		destinationCourse: Course
+	): Promise<CopyStatus[]> {
+		const promises: Promise<[number, CopyStatus]>[] = boardElements.map((element, pos) => {
+			if (element.boardElementType === BoardElementType.Task) {
+				const originalTask = element.target as Task;
+				return this.taskCopyService
+					.copyTask({
+						originalTask,
+						user,
+						destinationCourse,
+					})
+					.then((status) => [pos, status]);
+			}
+			if (element.boardElementType === BoardElementType.Lesson) {
+				const originalLesson = element.target as Lesson;
+				return this.lessonCopyService
+					.copyLesson({
+						originalLesson,
+						user,
+						destinationCourse,
+					})
+					.then((status) => [pos, status]);
+			}
+
+			/* istanbul ignore next */
+			this.logger.warn(`BoardCopyService unable to handle boardElementType.`);
+			/* istanbul ignore next */
+			return Promise.reject(new Error(`BoardCopyService unable to handle boardElementType.`));
```

Comment:
This can be tested and a new error in generally is danger. It is not mapped in any situation to a 500er, but in some. 
For this service it is catched i think, but it feel wrong to let stay the code.

Reply:
Code exists as this place could technically can be reached if BoardElementTypes gets extended, but this mehtod is not extended to take care for the new types.
Code exists only as linter otherwise disagrees with our code ;-)